### PR TITLE
Add parent.X substitution

### PR DESCRIPTION
**Pull request by lgirault**
Fixes coursier/coursier/issues/867

### CODE REVIEW

**Comment by alexarchambault** [comment]
Thanks @lgirault. Maybe a non regression test could be added for that, with the failing dependency of https://github.com/coursier/coursier/issues/867. A simple test like [this one](https://github.com/coursier/coursier/blob/8a02ee8f15261880d53540f2a3dddbdbae3c2e23/tests/shared/src/test/scala/coursier/test/CentralTests.scala#L786-L791) can be added. In order to get the metadata for the test, one should set the `FILL_CHUNKS=1` in the environment while running the test, like `FILL_CHUNKS=1 sbt testsJVM/test`. This should add files under `tests/shared/src/test/resources/resolutions` and `tests/metadata`. Those of `tests/metadata` are in a git submodule, so should be pushed to the [metadata repo](https://github.com/coursier/test-metadata) (open a PR there). It's not too straightforward overall, so I can look into it if you don't feel like handling that.

**Comment by lgirault** [comment]
Thank you very much for the feedback and the explanation. I think I've done what you told me but I'm not confortable with git submodule. Do I need to commit the changes performed on `tests/metadata` ? (I mean  the "symlink/ref" toward this [commit](coursier/test-metadata/pull/2/commits/d86c1e53bf6fa2fb0600d2bc949412cdf130492a) ? )

**Comment by alexarchambault** [comment]
About the submodule, a `git status` in the `coursier` directory should show `tests/metadata` as modified. Now that https://github.com/coursier/test-metadata/pull/2 is merged, from the `tests/metadata` dir, you should be able to checkout the latest master. It can then be added with a `git add tests/metadata` from the `coursier` dir, and committed along with the rest.

File: tests/shared/src/test/scala/coursier/test/CentralTests.scala [hunk 1]
```diff
@@ -790,6 +790,13 @@ abstract class CentralTests extends TestSuite {
       )
     }
 
+    'parentVersionInPom - {
+      * - resolutionCheck(
```
Comment (alexarchambault) [inline tests/shared/src/test/scala/coursier/test/CentralTests.scala]: (nitpicking) the `* - ` isn't necessary here, like in `groupIdVersionProperties` above. It just adds an extra level in the utest output (I see that `parentBeforeImports` does that below… but it shouldn't…).

**Comment by alexarchambault** [comment]
Merging, thanks!

**Comment by lgirault** [comment]
sweet ! thank you !

**Pull request merged**